tests/test_zlib.rs

```diff
@@ -0,0 +1,331 @@
+use anyhow::Context;
+use fs_extra::dir::{copy,CopyOptions};
+use serial_test::serial;  // for tests that change directories
+use std::fs::File;
+use std::io::Read;
+use std::path::{PathBuf, Path};
+use tempfile::{TempDir,tempdir};
+use xshell::{Cmd, pushd};
+
+use bom;
+use bom::bom::options::{BitcodeOptions,ExtractOptions};
+
+mod common;
+
+/// Runs a series of tests by using/building the zlib sources.  The zlib
+/// distribution is rather ideal for this because:
+///
+///  1. It is relatively small, but does consist of multiple sources
+///  2. It builds both shared and static libraries
+///  3. It builds sample executables, some statically linked, some share linked
+///
+/// These features allow a number of different verifications of build-bom to be
+/// performed here.
+
+// Download the file at the given URL to [tests/sources/<filename>] if it
+// doesn't already exist, returning the canonical absolute path to that file.
+fn fetch_if_needed(url : &str, filename : &str) -> anyhow::Result<PathBuf> {
+    let path = Path::new(common::SOURCE_DIR).join(filename);
+    if !path.exists() {
+        let cmd = Cmd::new("wget").arg("-O").arg(path.as_path()).arg(url);
+        cmd.run()?;
+    }
+
+    let abs_path = std::fs::canonicalize(path.as_path())?;
+    Ok(abs_path)
+}
+
+
+#[allow(dead_code)]  // tdir is holding a resource, but never directly used
```

Comment:
Can you just prefix its name with an underscore to get rid of that warning?

Reply:
This was actually obsolete because I do have an explicit reference in each of the tests, so I removed it.  The main concern is that it represents the tmpdir which will be removed when this is dropped, so it shouldn't be dropped early. I would have been worried that an underscore prefix and lack of reference would cause rust to drop it if optimized, but per the above, this isn't needed anymore.